### PR TITLE
feat: Update organization logo

### PR DESCRIPTION
**Pull request by gkats**
## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🌟 New feature (non-breaking change which adds functionality)
- [ ] 🔨 Breaking change (fix or feature that would cause existing functionality)
- [ ] 📖 Docs change / refactoring / dependency upgrade to change)

## Description

Added a new method on the organizations service that allows to set or update an organization's logo. 

Accepts the organization ID, the file for the logo and the uploader user ID as parameters. Triggers a multipart/form-data request to our Backend API to update the organization logo.

### CODE REVIEW

File: clerk/organizations.go [hunk 1]
```diff
@@ -66,6 +69,44 @@ func (s *OrganizationsService) Update(organizationID string, params UpdateOrgani
 	return &organization, nil
 }
 
+type UpdateOrganizationLogoParams struct {
+	File           multipart.File
+	Filename       string
```
Comment (gkats) [inline clerk/organizations.go:74]: I couldn't find an elegant way to get the filename from the file. If anybody has any suggestion, please help. :smile:

Comment (gkats) [reply clerk/organizations.go]: Made the parameter optional.

File: tests/integration/organizations_test.go [hunk 2]
```diff
@@ -86,6 +88,21 @@ func TestOrganizations(t *testing.T) {
 	}
 	assert.Equal(t, slug, *updatedOrganization.Slug)
 
+	// Update organization logo
+	assert.Nil(t, updatedOrganization.LogoURL)
+	filename := "200x200-grayscale.jpg"
+	file, err := os.Open(fmt.Sprintf("../../testdata/%s", filename))
```
Comment (georgepsarakis) [inline tests/integration/organizations_test.go:94]: [Minor] You can use [path.Join](https://pkg.go.dev/path#Join) to form the path.

File: clerk/organizations_test.go [hunk 3]
```diff
@@ -199,6 +202,80 @@ func TestOrganizationsService_ListAll_invalidServer(t *testing.T) {
...
+			if header.Filename != filename {
+				t.Errorf("expected %s, got %s", filename, header.Filename)
+			}
+			defer fileParam.Close()
+
+			var (
+				got  []byte
+				want []byte
+			)
+			_, err = fileParam.Read(got)
```
Comment (georgepsarakis) [inline clerk/organizations_test.go:244]: 🔧 I may be missing something here but the `Read` methods read number of bytes equal to the length of the slice that's passed to them. If we pass an empty slice then nothing is read I think.

File: clerk/organizations_test.go [hunk 4]
```diff
@@ -199,6 +202,80 @@ func TestOrganizationsService_ListAll_invalidServer(t *testing.T) {
...
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, err = file.Read(want)
```
Comment (georgepsarakis) [inline clerk/organizations_test.go:248]: ❓ Isn't the file handle already seeked to its end within the `UpdateLogo` function? I think you may need to seek to the start or re-open? 🤔

File: clerk/organizations_test.go (hunk 3 above)
Comment (gkats) [reply clerk/organizations_test.go]: Great catch, thanks! Updated the code to set the slice length to each file's size before reading.

File: clerk/organizations_test.go (hunk 4 above)
Comment (gkats) [reply clerk/organizations_test.go]: Updated to move reset the seeker.

File: tests/integration/organizations_test.go (hunk 2 above)
Comment (gkats) [reply tests/integration/organizations_test.go]: :+1: Updated here and the `clerk/organizations_test.go` file.

**Review by georgepsarakis — APPROVED** [review]

**Pull request merged**